rfc/router-storage.md

```diff
@@ -0,0 +1,767 @@
+      // as best I can tell this is used to determine who is the audience for the location commitment, which should be web3.storage?
+      // alternatively maybe it should be directly to space
+      "space": "did:web:web3.storage",
+      "blob": {
+        // multihash of the blob as byte array
+        "content": { "/": { "bytes": "mEi...sfKg" } },
+        "size": 2_097_152,
+      },
+      "expires": 1712903125,
+    }
+  }
+}
+```
+
+Receipt:
+
+```json
+{ 
+  // "/": "bafy..work",
+  "iss": "did:web:node1.storage",
+  "aud": "did:web:web3.storage",
+  "cmd": "/ucan/assert",
+  "sub": "did:web:web3.storage",
+  "args": {
+    "assert": [
+      // refers to the invocation from the example
+      { "/": "bafy..accept" },
+      // refers to the receipt corresponding to the above invocation
+      { 
+        "out": {
+          "ok": {
+            // result is commitment that is produced as result of "bafy..accept"
+            "site": "bafy..locationCommitment"
+          }
+        },
+        "next": []
+      },
+    ],
+  },
+}
+```
+
+Legacy receipt:
+
+```json
+{ 
+  // "/": "bafy..work",
+  "iss": "did:web:node1.storage",
+  "aud": "did:web:web3.storage",
+  "ran": {
+    "/": "bafy..accept"
+  },
+  "out": {
+    "ok": {
+      // result is commitment that is produced as result of "bafy..accept"
+      "site": "bafy..locationCommitment"
+    }
+  },
+  "fx": [],
+}
+```
+
+4. Router fetches location commitment, and creates new commitment for the user.
+
+**Note: initially I thought this was a redelegation, but I think it's maybe not -- the commitment here is to a limited download URL of JUST the blob -- it's not a w3s gateway url with full download capabilities.
+
+the "did:web:web3.storage" is maybe really a did for two seperate things:
+- the web3.storage read pipeline
+- the web3.storage write pipeline
+
+As long as we're the only ones running the read pipeline, it can all go through w3s.link, but I imagine a future situation where other people want to use another read pipeline provider.
+
+We also need to start thinking about the difference between a storage link -- i.e. a place to just download a CAR file as is with range parameters, and a gateway endpoint -- like w3s.link that serves flat files, pathing, etc -- even in an "encodingless IPFS" world, w3s.link is different that it's CDN cached and highly available **
+
+Original:
+
+```json
+{
+  "iss": "did:web:node1.storage",
+  "aud": "did:web:web3.storage",
+  "cmd": "/assert/location",
+  "sub": "did:web:node1.storage",
+  "pol": [
+    // multihash must match be for the blob uploaded
+    ["==", ".content", { "/": { "bytes": "mEi...sfKg" } }],
+    // must be available from this url -- but for a storage node this is probably just a raw blob, not w3s.link with full gateway support
+    ["==", ".url", "https://node1.storage/raw/bafy...7fi"],
+    // from this range
+    ["==", ".range[0]", 0],
+    ["==", ".range[1]", 2_097_152],
+  ],
+  // does not expire
+  "exp": null
```

Comment:
I think if we are in the middle we could issue non-expiring commitments and take on the task of renewing deals with storage nodes to uphold this commitment. On the other hand if we aren't in the middle then commitment really should have a expiry.